src/Common/Common.Api/Datasets/DatasetsClient.cs

```diff
@@ -76,12 +77,31 @@ public Table UpdateTable(Table table, Guid datasetId, Guid? workspaceId = null)
             return result as Table;
         }
 
-        public object AddRows(string datasetId, string tableName, JArray rows, Guid? workspaceId = default)
+        public object AddRows(string datasetId, string tableName, List<PSObject> rows, Guid? workspaceId = default)
         {
-            
+            var hashRows = new List<Hashtable>();
+            foreach (var row in rows)
+            {
+                var hashtable = new Hashtable();
+                if (row.BaseObject is Hashtable)
```

Comment:
You could declare variable name at the end then avoid the as below

Reply:
Sorry but i didnt get what you mean here. Could you add a bit more explanation?

Reply:
if (row.BaseObject is Hashtable ht) {
foreach (DictionaryEntry baseObj in ht) {
...

Reply:
Oh okay,  I did that before and variable name was a bit vague as there is already hashtable and I had to do like you did "ht". So I changed it to explicitly cast in the later code. But if you prefer it, I can change the code.

Reply:
We can leave it for now. I'm going to submit some doc changes and we will release later today.